### PR TITLE
Fix a bunch of unused variable / parameter shadowing warnings

### PR DESCRIPTION
**Pull request by atmyers**
Note that I didn't touch those originating from picsar.

### CODE REVIEW

File: Source/Parallelization/WarpXComm.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ WarpX::UpdateAuxilaryData ()
             MultiFab::Subtract(dBz, *Bfield_cp[lev][2], 0, 0, 1, ng);
 
             const Real* dx = Geom(lev-1).CellSize();
-            const int ref_ratio = refRatio(lev-1)[0];
+            const int rr = refRatio(lev-1)[0];
```
Comment (MaxThevenet) [inline Source/Parallelization/WarpXComm.cpp]: Do you think we should use a name more explicit, e.g., `refinement_ratio` or something like that?

Comment (atmyers) [reply Source/Parallelization/WarpXComm.cpp]: I changed this.

**Comment by atmyers** [comment]
I think that's a good convention, but we are very far from following it at the moment...

**Comment by MaxThevenet** [comment]
That's is true. We could also create an issue to make WarpX comply with its `style and conventions` rules, and finally use label `good first issue`, that GitHub keeps advertising... 
Anyway is fine with me, I'll merge when tests pass.

**Pull request merged**